### PR TITLE
[CMD] Properly handle case when command is given with leading or trailing spaces

### PR DESCRIPTION
**Pull request by zefklop**
For the testbot

JIRA issue: [CORE-17351](https://jira.reactos.org/browse/CORE-17351), [CORE-16898](https://jira.reactos.org/browse/CORE-16898), [CORE-17612](https://jira.reactos.org/browse/CORE-17612)

### CODE REVIEW

**Comment by HBelusca** [comment]
With Windows' CMD: `cmd /c "test program.bat   "` will indeed run the .bat, but `cmd /c "test program.battt   "` won't and will ask the user what to use in order to run the file. Does this patch make it behave that way?

File: base/shell/cmd/cmd.c [hunk 1]
```diff
@@ -417,7 +417,7 @@ Execute(LPTSTR Full, LPTSTR First, LPTSTR Rest, PARSED_COMMAND *Cmd)
 
     /* check if this is a .BAT or .CMD file */
     dot = _tcsrchr (szFullName, _T('.'));
-    if (dot && (!_tcsicmp (dot, _T(".bat")) || !_tcsicmp (dot, _T(".cmd"))))
+    if (dot && (!_tcsncicmp (dot, _T(".bat"), 4) || !_tcsncicmp (dot, _T(".cmd"), 4)))
```
Comment (HBelusca) [inline base/shell/cmd/cmd.c:420]: The same code in line 242 of start.c certainly needs to be patched too.

**Comment by HBelusca** [comment]
Now, test `cmd /c "test program.bat.    "` (there's a fullstop after `bat`) and.... enjoy 😂

**Comment by HBelusca** [comment]
Here's a test case:
File `test program.bat`:
```
@echo off
echo This is my BAT: '%0' params: '%*'
```
and make a copy of it with the name `test program.batx` and place it in the same directory.

Results: ("Windows": Windows' CMD: "ReactOS": ReactOS' CMD)
1. `cmd /c "test program.bat   "`
 Windows: OK: `This is my BAT: '"test program.bat   "' params: ''`
 ReactOS: OK (with PR) - Same result.
`cmd /c "test program.bat.   "` --> Windows: OK  |  ReactOS: Infinite loop
2. `cmd /c "test program.bat.   "`
 Windows: OK: `This is my BAT: '"test program.bat.   "' params: ''`
 ReactOS (with or without PR): Infinite loop.
3. cmd /c "test program.batx   "
 Windows: OK: Explorer will ask with which program to open this .batx file.
 ReactOS: Not OK, will run the batx file as a batch file: `This is my BAT: '"test program.batx   "' params: ''`
 (you can verify this by renaming or deleting the original .bat file, and keeping only the batx file).

**Comment by zefklop** [comment]
Now, GetFullPathName to the rescue.

**Pull request merged**